macros/print_base/park.cfg

```diff
@@ -4,24 +4,30 @@ gcode:
     {% set E = params.E|default(1.7)|float %}
 
     {% set Px, Py = printer["gcode_macro _USER_VARIABLES"].park_position_xy|map('float') %}
-    {% set max_z = printer.toolhead.axis_maximum.z|float %}
+    {% set park_lift_z = printer["gcode_macro _USER_VARIABLES"].park_lift_z %}
 
     {% set St = printer["gcode_macro _USER_VARIABLES"].travel_speed * 60 %}
     {% set Sz = printer["gcode_macro _USER_VARIABLES"].z_drop_speed * 60 %}
 
+    {% set max_z = printer.toolhead.axis_maximum.z|float %}
     {% set act_z = printer.toolhead.position.z|float %}
-    {% if act_z < (max_z - 50.0) %}
-        {% set z_safe = 50.0 %}
-    {% else %}
-        {% set z_safe = max_z - act_z %}
+
+    {% set z_safe = act_z + park_lift_z %}
+    {% if z_safe > max_z %}
+        {% set z_safe = max_z %}
     {% endif %}
 
+    # retract filament before move up toolhead
     {% if printer.extruder.temperature > 185 %}
-        G1 E-{E} F2100
+        G10
     {% endif %}
+    G90
+    G1 Z{z_safe} F{Sz}
 
+    {% if printer.extruder.temperature > 185 %}
     G91    
-    G1 Z{z_safe} F{Sz}
+        G1 E-{E} F2100
     G90
+    {% endif %}
```

Comment:
I'm OK to use G10/G11 here if you want.

But then, why having then another E-{E} ? I suspect it's for the case where we do PARK E=10 like at the end of a print ?
Then better to change the default E value to 0 in this case. So if no E parameter is put in the call, then there will be no "double retract"

Reply:
Ok, I've checked all the file. As we are also doing a manual E-10 in end_print and cancel_print, then we can remove it from here as you suggested. Can you do it ?

Reply:
> Ok, I've checked all the file. As we are also doing a manual E-10 in end_print and cancel_print, then we can remove it from here as you suggested. Can you do it ?

These retract are necessary to be able to manually remove the filament at the end/cancel print.
Might be a good idea to put that 10mm retract in _TIP_Shaping, right?

Reply:
Mhmm not a really fan of that as we could also just call `_TIP_SHAPING` for something else like for the ERCF or I don't know. That means the `_TIP_SHAPING` could not be associated everytime with a filament unload I think.

Reply:
Ok, so we need to keep `G0 E-10` in end/cancel print else we can't remove manually filament.